### ISSUE TITLE
make-collectd-graphs.sh should respect Lighttpd's www root directory

### ISSUE THREAD

**Issue by mgunther68** [jprochazka/adsb-feeder#42]
Looks like /var/www/html/ is currently hard-coded in make-collectd-graphs.sh

**Comment by jprochazka** [comment]
In too much of a rush last night did not even think to fix this file...

**Comment by jprochazka** [comment]
Taken care of.
Thanks again.

**Comment by mgunther68** [comment]
I tried on my Lubuntu VM and it does not seem to work. The script is not getting the document root directory and the variable stays blank.
Running the script manually I get this error for example:
`ERROR: Could not save png to '/graphs/system-localhost-cpu-365d.png'`

And when adding 2>/tmp/error to the end of the line in the cron file, I get in the error file:
`/home/marcus/adsb-feeder/build/portal/graphs/make-collectd-graphs.sh: line 4: lighttpd: command not found`

**Comment by jprochazka** [comment]
I have to run to the store right now but when I get back I will look into this. I take Lighttpd is installed. Wonder if it is not in the path by default on that distribution version.

**Comment by mgunther68** [comment]
Yes the other ones are working fine and Lighttpd is installed:
```
marcus@lubuntu:~/adsb-feeder$ which lighttpd
/usr/sbin/lighttpd
```
This is very strange. Let me try on the Pi and report back.

**Comment by mgunther68** [comment]
Not working on the Pi either I'm afraid.
```
pi@raspberrypi ~/git/adsb-feeder $ sudo bash /home/pi/git/adsb-feeder/build/portal/graphs/make-collectd-graphs.sh 1h >/dev/null
ERROR: Could not save png to '/graphs/dump1090-localhost-aircraft-1h.png'
ERROR: Could not save png to '/graphs/dump1090-localhost-cpu-1h.png'
ERROR: Could not save png to '/graphs/dump1090-localhost-tracks-1h.png'
ERROR: opening '/var/lib/collectd/rrd/localhost/aggregation-cpu-average/cpu-idle.rrd': No such file or directory
ERROR: Could not save png to '/graphs/system-localhost-df_root-1h.png'
ERROR: Could not save png to '/graphs/system-localhost-disk_io_iops-1h.png'
ERROR: Could not save png to '/graphs/system-localhost-disk_io_octets-1h.png'
ERROR: Could not save png to '/graphs/system-localhost-eth0_bandwidth-1h.png'
ERROR: Could not save png to '/graphs/system-localhost-memory-1h.png'
ERROR: Could not save png to '/graphs/system-localhost-temperature-1h.png'
ERROR: Could not save png to '/graphs/dump1090-localhost-local_rate-1h.png'
ERROR: Could not save png to '/graphs/dump1090-localhost-local_trailing_rate-1h.png'
ERROR: Could not save png to '/graphs/dump1090-localhost-range-1h.png'
ERROR: Could not save png to '/graphs/dump1090-localhost-signal-1h.png'
```

And without the 'bash':
```
pi@raspberrypi ~/git/adsb-feeder $ sudo /home/pi/git/adsb-feeder/build/portal/graphs/make-collectd-graphs.sh 1h >/dev/null
/home/pi/git/adsb-feeder/build/portal/graphs/make-collectd-graphs.sh: 1: /home/pi/git/adsb-feeder/build/portal/graphs/make-collectd-graphs.sh: Syntax error: redirection unexpected
```

**Comment by mgunther68** [comment]
OK.. two issues I think:
1) cron does not have the full path variable:
`* * * * * root echo $PATH > /tmp/path` gives me:
```
pi@raspberrypi ~/git/adsb-feeder $ cat /tmp/path
/usr/bin:/bin
```
But lighttpd is in /usr/sbin/

So, adding
`PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin`
to the cron file works for me.
(see http://stackoverflow.com/questions/10129381/crontab-path-and-user)

b) the renice -n 5 -p $$ seems out of place. It works for me when I put it in a separate line after the variables are assigned.

**Comment by jprochazka** [comment]
I hardcode the path to lighttpd as /usr/sbin/lighttpd. The path appears to be the same in both Wheezy and Jessie distributions. This change was added in commit 6e2be44de380bc4893065fdd232d6a55f686619b. I am doing this in case the $PATH variable for the user does not contain /usr/sbin as well. If there is an issue with hardcoding the paths down the road the path can be added to the cron.

I placed the renice line in it's proper place in commit e798e4050c070b29cbe43a22b788aa8731da0268.
